qiskit/providers/aer/pulse_new/models/operator_models.py

```diff
@@ -0,0 +1,409 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+from typing import List
+import numpy as np
+
+from .signals import VectorSignal, Signal, Constant
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+from qiskit.quantum_info.operators import Operator
+
+class OperatorModel:
+    """OperatorModel representing a sum of :class:`Operator` objects with
+    time dependent coefficients.
+
+    Specifically, this object represents a time dependent matrix of
+    the form:
+
+    .. math::
+
+        G(t) = \sum_{i=0}^{k-1} s_i(t) G_i,
+
+    for :math:`G_i` matrices (represented by :class:`Operator` objects),
+    and the :math:`s_i(t)` given by signals represented by a
+    :class:`VectorSignal` object, or a list of :class:`Signal` objects.
+
+    This object contains functionality to evaluate :math:`G(t)` for a given
+    :math:`t`, or to compute products :math:`G(t)A` and :math:`AG(t)` for
+    :math:`A` an :class:`Operator` or array of suitable dimension.
+
+    Additionally, this class has functionality for representing :math:`G(t)`
+    in a rotating frame, and setting frequency cutoffs
+    (e.g. for the rotating wave approximation).
+    Specifically, given an anti-Hermitian frame operator :math:`F` (i.e.
+    :math:`F^\dagger = -F`), entering the frame of :math:`F` results in
+    the object representing the operator :math:`e^{-Ft}G(t)e^{Ft} - F`.
+
+    Further, if a frequency cutoff is set, when evaluating the
+    `OperatorModel`, any terms with a frequency above the cutoff
+    (which combines both signal frequency information and frame frequency
+    information) will be set to :math:`0`.
+    """
+
+    def __init__(self,
+                 operators,
+                 signals,
+                 signal_mapping=None,
+                 frame_operator=None,
+                 cutoff_freq=None):
+        """Initialize.
+
+        Args:
+            operators (list): list of Operator objects.
+            signals (list): Specifiable as either a VectorSignal, a list of
+                            Signal objects, or as the inputs to signal_mapping.
+            signal_mapping (Callable): a function returning wither a
+                                       VectorSignal or a list of Signal objects.
```

Comment:
This docstring could greatly benefit from an example. Looking at the signals setter below it is not quite clear how this mapping should work.

Reply:
I've added an example into the demo as well as in the class doc string for how this is meant to behave.